web/themes/custom/novel/novel.theme

```diff
@@ -35,6 +38,25 @@ function novel_theme(
   ];
 }
 
+/**
+ * Implements hook_preprocess_HOOK().
+ *
+ * @param mixed[] $variables
+ *   The variables for the theme hook.
+ */
+function novel_preprocess_dpl_react_app__search_header(array &$variables): void {
+  $search_svg = file_get_contents(sprintf('%s/%s',
```

Comment:
So the plan is to inline all svg icons etc. or is this some skeleton images?

Reply:
When using `file_get_contents` you should check that it actually gets data and not is `false`.

Reply:
How much is the performance gain by doing this with svg icons? And why is it only this svg icon?

Reply:
> So the plan is to inline all svg icons etc. or is this some skeleton image?

This is to be able to render the search header fast. It is the same principle of showing the rendering server side before javascript kicks in with it's own rendering. In this particular case we show the search icon (and the search field) before the dpl-react app comes and does the same.

Reply:
> When using `file_get_contents` you should check that it actually gets data and not returns `false`.

Good point. I'll look into that.

Reply:
Quick note: `thecodingmachine/safe` ensures that `file_get_contents` will throw an exception if loading fails - not return `false`.